graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -767,14 +768,18 @@ public void revokeToken(
     }
 
     @VisibleForTesting
-    boolean isTokenCreationAllowed(User user) {
-        final boolean allowed = isPermitted(USERS_TOKENCREATE, user.getName());
-        final boolean isAdmin = isAdmin(user);
+    boolean isTokenCreationAllowed(User callingUser, User futureOwner) {
+        final boolean allowed = isPermitted(USERS_TOKENCREATE, callingUser.getName());
+        final boolean isAdmin = isAdmin(callingUser);
         if (isAdmin) {
             return allowed;
         }
-        final boolean externalAllowed = isExternalUserAllowed(user);
-        final boolean adminAllowed = isAllowedAsNoAdmin(user);
+        if (!Objects.equals(callingUser.getId(), futureOwner.getId())) {
+            // Only admins are allowed to create tokens for other users, but we already checked for admin above.
+            return false;
+        }
+        final boolean externalAllowed = isExternalUserAllowed(callingUser);
```

Comment:
this one should be the future owner, afaik admin can create tokens even when there are external but regular users should not be allowed to. not sure how likely it is that an admin will create a token for another admin

Reply:
just synced with Florian. for now admin will be allowed to create tokens for external. so we are just checking if the current user is an external user and is allowed to create tokens